### PR TITLE
Update NFT Server

### PR DESCRIPTION
**Pull request by hochiw**
Postman collection
[AuthGear NFT Indexer API Server.postman_collection.json.zip](https://github.com/authgear/authgear-nft-indexer/files/9782207/AuthGear.NFT.Indexer.API.Server.postman_collection.json.zip)

refs #34

### CODE REVIEW

File: pkg/model/database/nft_ownership.go [hunk 1]
```diff
@@ -39,8 +45,27 @@ func (c NFTOwnership) ToAPIToken() apimodel.Token {
...
 			Index:     *c.BlockNumber.ToMathBig(),
-			Timestamp: *c.BlockTimestamp,
+			Timestamp: c.BlockTimestamp,
 		},
 		Balance: c.Balance,
 	}
 }
+
+func (c NFTOwnership) IsEmpty() bool {
+	return c.Balance == "0" || c.TokenID == "0x0" || c.BlockNumber == bunbig.FromInt64(0) || c.TransactionHash == "0x0" || c.TransactionIndex == 0 || c.BlockTimestamp == nil
```
Comment (louischan-oursky) [inline pkg/model/database/nft_ownership.go:54]: Just check c.TransactionHash == "0x0" is enough.

File: pkg/handler/get_metadata.go [hunk 2]
```diff
@@ -29,25 +24,14 @@ func NewGetCollectionMetadataHandlerLogger(lf *log.Factory) GetCollectionMetadat
...
 }
 
 type GetCollectionMetadataAPIHandler struct {
-	JSON                 JSONResponseWriter
-	Logger               GetCollectionMetadataHandlerLogger
-	AlchemyAPI           GetCollectionMetadataHandlerAlchemyAPI
-	NFTCollectionQuery   query.NFTCollectionQuery
-	NFTCollectionMutator GetCollectionMetadataNFTCollectionMutator
-	RateLimiter          GetCollectionMetadataRateLimiter
+	JSON            JSONResponseWriter
```
Comment (louischan-oursky) [inline pkg/handler/get_metadata.go:33]: RateLimit should still be applied in the handler, not in the service.

File: pkg/handler/probe_collection.go [hunk 3]
```diff
@@ -27,28 +24,14 @@ func NewProbeCollectionHandlerLogger(lf *log.Factory) ProbeCollectionHandlerLogg
...
-	InsertNFTCollectionProbe(contractID authgearweb3.ContractID, isLargeCollection bool) (*database.NFTCollectionProbe, error)
-}
 type ProbeCollectionAPIHandler struct {
-	JSON                      JSONResponseWriter
-	Logger                    ProbeCollectionHandlerLogger
-	AlchemyAPI                ProbeCollectionHandlerAlchemyAPI
-	RateLimiter               ProbeCollectionHandlerRateLimiter
-	NFTCollectionProbeQuery   ProbeCollectionHandlerNFTCollectionProbeQuery
-	NFTCollectionProbeMutator ProbeCollectionHandlerNFTCollectionProbeMutator
+	JSON         JSONResponseWriter
```
Comment (louischan-oursky) [inline pkg/handler/probe_collection.go:32]: Keep rate limit at handler.

File: pkg/query/nft_ownership.go [hunk 4]
```diff
@@ -58,6 +59,12 @@ func (b NFTOwnershipQueryBuilder) WithOwner(ownerID *authgearweb3.ContractID) NF
 	}
 }
 
+func (b NFTOwnershipQueryBuilder) WithExpiry(duration time.Duration) NFTOwnershipQueryBuilder {
+	return NFTOwnershipQueryBuilder{
+		b.Where("(created_at + '? second') > NOW()", duration.Seconds()),
+	}
+}
```
Comment (louischan-oursky) [inline pkg/query/nft_ownership.go:66]: Change to this

```golang
func (b NFTOwnershipQueryBuilder) WithMinimumFreshness(t time.Time) NFTOwnershipQueryBuilder {
  return NFTOwnershipQueryBuilder{
    b.Where("created_at > ?", t),
  }
}
```

and t should be something like

```golang
t := clock.NowUTC()
t = t.Add(-n * time.Seconds)
```

**Comment by hochiw** [comment]
@louischan-oursky updated :adore2x:

**Pull request merged**